hls/lib/hls/compositor_handler.ex

```diff
@@ -56,104 +57,66 @@ if Code.ensure_loaded?(Membrane.VideoCompositor) do
     def handle_inputs_change(
           inputs,
           _ctx,
-          state
-        ) do
-      inputs_number = Enum.count(inputs)
+          %{output_stream_format: output_stream_format} = state
+        )
+        when map_size(inputs) > 0 do
+      inputs_amount = Enum.count(inputs)
+
+      max_inputs_in_row = :math.sqrt(inputs_amount) |> ceil()
+      row_amount = ceil(inputs_amount / max_inputs_in_row)
+
+      output_row_height = div(output_stream_format.height, row_amount)
+      output_col_width = div(output_stream_format.width, max_inputs_in_row)
+
+      desired_stream_format = %{
+        width: output_col_width - @padding * 2,
+        height: output_row_height - @padding * 2
+      }
```

Comment:
There are some parts of this function that can be moved into separate functions. What do you think?

```suggestion
{max_inputs_per_row, rows_count} = calculate_grid(inputs_count)

    {desire_format_width, desire_format_height} = calculate_desired_format(output_format, max_inputs_per_row, rows_count)
```

Reply:
Yeah, but I'm using some of these intermediates like `output_(row|col)_(height|width)` inside the map functions...

You can make a commit with your suggested changes if you have the time :wink:

Reply:
sorry I didn't notice that